sdks/python/apache_beam/runners/worker/bundle_processor.py

```diff
@@ -846,6 +870,30 @@ def process_bundle(self, instruction_id):
         data_channels[input_op.data_channel].append(input_op.transform_id)
         input_op_by_transform_id[input_op.transform_id] = input_op
 
+        # Set up timer output stream
+      timer_output_streams = {}
+      for transform_id, timer_list in self.timer_ids.items():
+        output_streams = {}
+        for timer_id in timer_list:
+          output_streams[
+              timer_id] = self.timer_data_channel.output_timer_stream(
+                  instruction_id, transform_id, timer_id)
+          timer_output_streams[transform_id] = output_streams
+        self.process_timer_ops[
+            transform_id].user_state_context.update_timer_output_streams(
+                output_streams)
+
+      # Process timers
+      if self.timer_data_channel:
```

Comment:
We can't safely assume the runner will finish sending all timers before sending any of the data (and the buffer may get full, resulting in a deadlock). I think we need to have a data_channel.inputs() that returns both data and timers and then branch in the loop.